Networking/WebSockets/WebSocketImpl.cc

```diff
@@ -85,18 +85,23 @@ namespace litecore { namespace websocket {
     }
 
 
+#define CHECK_DELEGATE { if (!hasDelegate()) return; }
+
+
     void WebSocketImpl::connect() {
         logInfo("Connecting...");
         startResponseTimer(chrono::seconds(kConnectTimeoutSecs));
     }
 
 
     void WebSocketImpl::gotHTTPResponse(int status, const websocket::Headers &headersFleece) {
+        CHECK_DELEGATE
         logInfo("Got HTTP response (status %d)", status);
         delegate().onWebSocketGotHTTPResponse(status, headersFleece);
     }
 
     void WebSocketImpl::onConnect() {
+        CHECK_DELEGATE
```

Comment:
This method does a lot of stuff besides call the delegate. Why would you make it return and do nothing, when the delegate is null?